File: server/src/Bananagrams.Dal/Models/GameUserGameAnagram.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Diagnostics.CodeAnalysis;

namespace Bananagrams.Dal.Models;

[Table("game_user_game_anagrams")]
public class GameUserGameAnagram
{
    [Key]
    [Column("id")]
    public int Id { get; set; }
    [Column("attempts")]
    public int Attempts { get; set; }
    [Column("date_played")]
    public DateTime? DatePlayed { get; set; }
    [Column("date_solved")]
    public DateTime? DateSolved { get; set; }
    [Column("game_user_id")]
    public int GameUserId { get; set; }
    [Column("game_anagram_id")]
    public int GameAnagramId { get; set; }
    public GameUser GameUser { get; set; }    
    public GameAnagram GameAnagram { get; set; }
}

File: server.java/bananagrams/src/main/java/coe/bananagrams/entity/GameUserGameAnagram.java
package coe.bananagrams.entity;

import jakarta.persistence.*;
import lombok.*;

import java.util.Date;

@Getter
@Setter
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "game_user_game_anagrams")
public class GameUserGameAnagram {
    @Id
    @Column(name = "id", updatable = false, nullable = false)
    private int id;
    @Column(name = "date_played", nullable = true)
    private Date datePlayed;
    @Column(name = "date_solved", nullable = true)
    private Date dateSolved;
    @Column(name = "game_user_id", insertable = false, updatable = false, nullable = false)
    private int gameUserId;
    @Column(name = "game_anagram_id", insertable = false, updatable = false, nullable = false)
    private int gameAnagramId;

    @ManyToOne()
    private GameUser gameUser;
    @ManyToOne()
    private GameAnagram gameAnagram;
}
